sql/core/src/main/scala/org/apache/spark/sql/util/ContinuousQueryListener.scala

```diff
@@ -65,11 +65,23 @@ object ContinuousQueryListener {
   trait Event
 
   /** Event representing the start of a query */
-  class QueryStarted private[sql](val query: ContinuousQuery) extends Event
+  class QueryStarted private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /** Event representing any progress updates in a query */
-  class QueryProgress private[sql](val query: ContinuousQuery) extends Event
+  class QueryProgress private[sql](val queryInfo: ContinuousQueryInfo) extends Event
 
   /** Event representing that termination of a query */
-  class QueryTerminated private[sql](val query: ContinuousQuery) extends Event
+  class QueryTerminated private[sql](val queryInfo: ContinuousQueryInfo) extends Event
```

Comment:
Should these be different events?  Its a little odd that `Progress` includes and `exception` and `Terminated` includes `isActive`.  We should consider what is the minimal amount of data we can expose here such that you can do all the useful things we need to.